lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift

```diff
@@ -0,0 +1,287 @@
+import SwiftSyntax
+import SwiftSyntaxMacros
+import SwiftDiagnostics
+
+public struct DebugDescriptionMacro {}
+
+extension DebugDescriptionMacro: PeerMacro {
+    public static func expansion<Decl, Context>(
+        of node: AttributeSyntax,
+        providingPeersOf declaration: Decl,
+        in context: Context
+    )
+    throws -> [DeclSyntax]
+    where Decl: DeclSyntaxProtocol, Context: MacroExpansionContext
+    {
+        // Determine the module name by doing a bit of parsing on fileID.
+        guard let fileID = context.location(of: declaration)?.file,
+              let moduleName = extractModuleName(from: fileID) else {
+            assertionFailure("could not determine module name from fileID")
+            return []
+        }
+
+        guard let typeName = identifier(of: declaration)?.text else {
+            let message: ErrorMessage = "must be attached to a struct/class/enum/extension"
+            context.diagnose(node: node, message: message)
+            return []
+        }
+
+        // Create a lookup for finding properties by name.
+        var properties: [String: PatternBindingSyntax] = [:]
```

Comment:
Should this property be `private` and added in another place like 
`// MARK: Variables `
`private var properties: [String: PatternBindingSyntax] = [:]`

Reply:
This is a local variable.